tensorflow/python/keras/metrics.py

```diff
@@ -2856,3 +2856,325 @@ def get(identifier):
   else:
     raise ValueError('Could not interpret '
                      'metric function identifier: %s' % identifier)
+
+class FBetaScore(Metric):
+    """Computes F-Beta Score.
+
+    This is the weighted harmonic mean of precision and recall.
+    Output range is [0, 1].
+
+    F-Beta = (1 + beta^2) * ((precision * recall) /
+                             ((beta^2 * precision) + recall))
+
+    `beta` parameter determines the weight given to the
+    precision and recall.
+
+    `beta < 1` gives more weight to the precision.
+    `beta > 1` gives more weight to the recall.
+    `beta == 1` gives equal weight to precision and recall.
+
+    Args:
+       num_classes : Number of unique classes in the dataset.
+       average : Type of averaging to be performed on data.
+                   Acceptable values are None, micro, macro and
+                   weighted.
+       beta : float. Determines the weight of precision and recall
+                in harmonic mean. Acceptable values are either a number
+                of float data type greater than 0.0 or a scale tensor
+                of dtype tf.float32.
+
+    Returns:
+       F Beta Score: float
+
+    Raises:
+       ValueError: If the `average` has values other than
+       [None, micro, macro, weighted].
+
+       ValueError: If the `beta` value is less than or equal
+       to 0.
+
+    `average` parameter behavior:
+
+    1. If `None` is specified as an input, scores for each
+       class are returned.
+
+    2. If `micro` is specified, metrics like true positivies,
+       false positives and false negatives are computed
+       globally.
+
+    3. If `macro` is specified, metrics like true positivies,
+       false positives and false negatives are computed for
+       each class and their unweighted mean is returned.
+       Imbalance in dataset is not taken into account for
+       calculating the score
+
+    4. If `weighted` is specified, metrics are computed for
+       each class and returns the mean weighted by the
+       number of true instances in each class taking data
+       imbalance into account.
+
+    Usage:
+    ```python
+    actuals = tf.constant([[1, 1, 0],[1, 0, 0]],
+              dtype=tf.int32)
+    predis = tf.constant([[1, 0, 0],[1, 0, 1]],
```

Comment:
In general there are a lot of formatting issues, please wrap to next line if > 80 chars and indent continuation line using 4 spaces.